### PR TITLE
doc: terminology cleanup in DM HLD

### PR DESCRIPTION
**Pull request by amyreye**
- Replace SOS or Service OS with Service VM
- Replace UOS or User OS with User VM
- Clean up some of the grammar

Signed-off-by: Amy Reyes <amy.reyes@intel.com>

### CODE REVIEW

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 1]
```diff
@@ -473,7 +475,7 @@ The following table summarizes the emulation of accesses to I/O port
 I/O Client Interfaces
 =====================
 
-.. note:: replace with reference to API documentation
+.. note:: Replace with reference to API documentation.
```
Comment (amyreye) [inline doc/developer-guides/hld/hld-devicemodel.rst:478]: Is this an internal note, saying to replace this section with API doc?

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 2]
```diff
@@ -502,33 +504,33 @@ Device Emulation
...
 
-A PIO range structure in DM is like below, it's the parameter needed to
-register PIO handler for special PIO range:
+A PIO range structure in the DM is shown below. It's the parameter needed to
+register a PIO handler for a special PIO range:
 
-.. note:: this should be references to API documentation in
-   devicemodel/include/inout.h
+.. note:: This should be references to API documentation in
+   ``devicemodel/include/inout.h``.
```
Comment (amyreye) [inline doc/developer-guides/hld/hld-devicemodel.rst:533]: Is this an internal note, saying to reference the API doc?

File: doc/developer-guides/hld/hld-devicemodel.rst (hunk 1 above)
Comment (dbkinder) [reply doc/developer-guides/hld/hld-devicemodel.rst:478]: This should reference our API documentation.  I can find these functions in the 2.0 docs (https://projectacrn.github.io/2.0/api/GVT-g_api.html) but they're gone in the current release (I think GVT-D was remove in v2.6).  @NanlinXie @gvancuts can you help here: should this section be deleted now?

File: doc/developer-guides/hld/hld-devicemodel.rst (hunk 2 above)
Comment (dbkinder) [reply doc/developer-guides/hld/hld-devicemodel.rst:533]: Yes, we shouldn't write API documentation like this, but should be extracting it from the doxygen comments.  Alas, there aren't any doxygen comments in inout.h as there should be.  (This is a note I wrote over three years ago that never got resolved.)

**Review by dbkinder — APPROVED** [review]
LGTM.  We should create an issue for the missing API comments.

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 3]
```diff
@@ -25,33 +25,33 @@ framework. There are 3 major subsystems in Service VM:
...
+The ``acrn-dm`` runs using these command-line configuration
 options:
 
 .. code-block:: none
 
-  acrn-dm [-hAWYv] [-B bootargs] [-E elf_image_path]
+   acrn-dm [-hAWYv] [-B bootargs] [-E elf_image_path]
                [-G GVT_args] [-i ioc_mediator_parameters] [-k kernel_image_path]
                [-l lpc] [-m mem] [-r ramdisk_image_path]
                [-s pci] [-U uuid] [--vsbl vsbl_file_name] [--ovmf ovmf_file_path]
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst:57]: We have removed `UUID` in ACRN, now may be a good time to also remove it from this doc.

```suggestion
               [-s pci] [--vsbl vsbl_file_name] [--ovmf ovmf_file_path]
```

There is another such instance further below.

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 4]
```diff
@@ -140,48 +140,49 @@ DM Initialization
 
 -  **DM Start**: DM application starts to run.
 
--  **Option Parsing**: DM parse options from command line inputs.
+-  **Option Parsing**: DM parses options from command-line inputs.
 
--  **VM Create**: DM calls ioctl to Service VM HSM, then Service VM HSM makes
-   hypercalls to HV to create a VM, it returns a vmid for a
+-  **VM Create**: DM calls ioctl to the Service VM HSM, then the Service VM HSM
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst:145]:
```suggestion
-  **VM Create**: DM calls ``ioctl`` to the Service VM HSM, then the Service VM HSM
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 5]
```diff
@@ -140,48 +140,49 @@ DM Initialization
...
+   makes hypercalls to the hypervisor to create a VM. It returns a vmid for a
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst:146]:
```suggestion
   makes hypercalls to the hypervisor to create a VM. It returns a VM ID (``vmid``) for a
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 6]
```diff
@@ -140,48 +140,49 @@ DM Initialization
...
    dedicated VM.
 
--  **Set I/O Request Buffer**: the I/O request buffer is a page buffer
-   allocated by DM for a specific VM in user space. This buffer is
-   shared between DM, HSM and HV. **Set I/O Request buffer** calls
+-  **Set I/O Request Buffer**: The I/O request buffer is a page buffer
+   allocated by the DM for a specific VM in user space. This buffer is
+   shared among the DM, HSM, and hypervisor. **Set I/O Request Buffer** calls
    an ioctl executing a hypercall to share this unique page buffer
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst:152]:
```suggestion
   an ``ioctl`` executing a hypercall to share this unique page buffer
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 7]
```diff
@@ -291,19 +292,19 @@ HSM
...
 ============
 
-Device Model manages User VM by accessing interfaces exported from HSM
-module. HSM module is a Service VM kernel driver. The ``/dev/acrn_hsm`` node is
-created when HSM module is initialized. Device Model follows the standard
-Linux char device API (ioctl) to access the functionality of HSM.
+The Device Model manages a User VM by accessing interfaces exported from the HSM
+module. The HSM module is a Service VM kernel driver. The ``/dev/acrn_hsm``
+node is created when the HSM module is initialized. The Device Model follows
+the standard Linux char device API (ioctl) to access HSM functionality.
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst:298]:
```suggestion
the standard Linux char device API (``ioctl``) to access the HSM functionality.
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 8]
```diff
@@ -291,19 +292,19 @@ HSM
...
 
-In most of ioctl, HSM converts the ioctl command to a corresponding
+In most of ioctl, the HSM converts the ioctl command to a corresponding
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst:300]:
```suggestion
In most ``ioctl``, the HSM converts the ``ioctl`` command to a corresponding
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 9]
```diff
@@ -818,8 +822,8 @@ Functions implemented by ACPI include:
 -  Battery management
 -  Thermal management
 
-All critical functions depends on ACPI tables.
-On an APL platform with Linux installed we can see these tables using:
+All critical functions depend on ACPI tables.
+On an Apollo Lake platform with Linux installed, we can see these tables using:
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst:826]: Appolo Lake is no longer supported. Can we update this to show the output on Tiger Lake
/cc @NanlinXie

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 10]
```diff
@@ -890,24 +894,24 @@ We can modify DMAR.dsl and assemble it again to AML:
...
 provided at a known place in the system low address space, and pointing
 to  an XSDT (Extended System Description Table). The following picture
-shows a typical ACPI table layout in an Intel APL platform:
+shows a typical ACPI table layout in an Apollo Lake platform:
 
 .. figure:: images/dm-image36.png
    :align: center
 
-   Typical ACPI table layout on Intel APL platform
+   Typical ACPI Table Layout on Apollo Lake Platform
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst:906]: Same here, we should replace this with the equivalent from a Tiger Lake system

File: doc/developer-guides/hld/hld-devicemodel.rst (hunk 9 above)
Comment (amyreye) [reply doc/developer-guides/hld/hld-devicemodel.rst:826]: > Appolo Lake is no longer supported. Can we update this to show the output on Tiger Lake /cc @NanlinXie

Suggest we create an issue ticket for future.

File: doc/developer-guides/hld/hld-devicemodel.rst (hunk 2 above)
Comment (amyreye) [reply doc/developer-guides/hld/hld-devicemodel.rst:533]: I created an issue.

File: doc/developer-guides/hld/hld-devicemodel.rst (hunk 1 above)
Comment (amyreye) [reply doc/developer-guides/hld/hld-devicemodel.rst:478]: If I delete this section, there are 2 references to it in this file at lines 396 and 702. Please advise on how to edit those lines.

**Comment by amyreye** [comment]
@dbkinder I accidently deleted the branch, then resurrected it. Now not sure why the "make doc" check is failing. I can build locally.

File: doc/developer-guides/hld/hld-devicemodel.rst (hunk 9 above)
Comment (gvancuts) [reply doc/developer-guides/hld/hld-devicemodel.rst:825]: OK, that works for me! (Tracked on #6861 )

File: doc/developer-guides/hld/hld-devicemodel.rst (hunk 1 above)
Comment (gvancuts) [reply doc/developer-guides/hld/hld-devicemodel.rst:477]: I don't know to be honest, @NanlinXie , can you get someone from the engineering team to clarify what we should do with these APIs. They did not _seem_ GVT-g specific but it's the only place we could find them and they seem gone. Have they been replaced by other APIs?

Comment (liuhang-bit) [reply doc/developer-guides/hld/hld-devicemodel.rst:477]: For I/O client registration

acrn_ioreq_create_client - create ioreq client

acrn_ioreq_add_iorange - add iorange monitored by ioreq client

acrn_ioreq_intercept_bdf - set intercept bdf info of ioreq client

acrn_ioreq_get_reqbuf - get request buffer

I/O client runtime helpers.

acrn_ioreq_attach_client - start handle request for ioreq client

acrn_ioreq_complete_request - notify guest request handling is completed

For I/O client destruction

acrn_ioreq_destroy_client - destroy ioreq client

acrn_ioreq_del_iorange - del iorange monitored by ioreq client

acrn_ioreq_unintercept_bdf - clear intercept bdf info of ioreq client

they are device model API , not GVT-g API , need to add it to device model API page

Comment (gvancuts) [reply doc/developer-guides/hld/hld-devicemodel.rst:477]: > they are device model API , not GVT-g API , need to add it to device model API page

They are not. None of these exist in the current codebase. What have they been replaced by?

There are a few that _resemble_ those here: https://github.com/projectacrn/acrn-hypervisor/blob/master/devicemodel/include/vmmapi.h#L98-L102. Is that it?

**Comment by dbkinder** [comment]
> @dbkinder I accidently deleted the branch, then resurrected it. Now not sure why the "make doc" check is failing. I can build locally.

@amyreye Problem is when the CI is trying to apply this patch:
```
+ log 'Start to git am https://github.com/projectacrn/acrn-hypervisor/pull/6841.patch'
+ [[ -n --ignore-whitespace ]]
+ git am --ignore-whitespace 6841.patch
+ log 'git am https://github.com/projectacrn/acrn-hypervisor/pull/6841.patch failed!'
+ log 'Start to abort failed patch'
```

At this point it would be easiest (for us) to close this PR and start a new one since you've got the local files building OK.  We'll lose the comments on the new PR, but we can refer the new PR to this PR #6841.

File: doc/developer-guides/hld/hld-devicemodel.rst (hunk 1 above)
Comment (amyreye) [reply doc/developer-guides/hld/hld-devicemodel.rst:477]: @liuhang-bit Please double-check existence of parameters. Cannot find them in code in repo. See Geoffroy's comment.

**Comment by gvancuts** [comment]
> > @dbkinder I accidently deleted the branch, then resurrected it. Now not sure why the "make doc" check is failing. I can build locally.
> 
> @amyreye Problem is when the CI is trying to apply this patch:
> 
> ```
> + log 'Start to git am https://github.com/projectacrn/acrn-hypervisor/pull/6841.patch'
> + [[ -n --ignore-whitespace ]]
> + git am --ignore-whitespace 6841.patch
> + log 'git am https://github.com/projectacrn/acrn-hypervisor/pull/6841.patch failed!'
> + log 'Start to abort failed patch'
> ```
> 
> At this point it would be easiest (for us) to close this PR and start a new one since you've got the local files building OK. We'll lose the comments on the new PR, but we can refer the new PR to this PR #6841.

You have additional commits in your branch, inc. one that reverts a previous commit. I suspect this error would go away if you could rebase your branch on top of master and squash all these commits.

**Comment by gvancuts** [comment]
> You have additional commits in your branch, inc. one that reverts a previous commit. I suspect this error would go away if you could rebase your branch on top of master and squash all these commits.

Hi @amyreye , I looked at your branch, removed the extra scruff and only kept the very last commit you had (which, upon inspection, seemed to have all the material for this PR). I applied that commit on top of `origin/master` and also abused my superpower by force-pushing this new branch to your repo, which automatically updated this PR as well.

I believe this should fix our build issues. Please also do a scan through the changes to make sure nothing was dropped in the process.

**Comment by gvancuts** [comment]
@wenlingz @NanlinXie , can we re-run our CI on this again to verify all is OK now?

**Comment by wenlingz** [comment]
> @wenlingz @NanlinXie , can we re-run our CI on this again to verify all is OK now?

re-run already PASS, CI auto-trigger once you update commit

File: doc/developer-guides/hld/hld-devicemodel.rst (hunk 1 above)
Comment (liuhang-bit) [reply doc/developer-guides/hld/hld-devicemodel.rst:478]: yes , the latest API is just like in the link above, so we should remove the obsolete one

Comment (gvancuts) [reply doc/developer-guides/hld/hld-devicemodel.rst:478]: > 
Can you provide for each of the APIs we need to include a brief description? I don't think Amy has yet quite enough info for this section.

**Comment by dbkinder** [comment]
@gvancuts Will you approve this PR as is now?

**Comment by gvancuts** [comment]
> @gvancuts Will you approve this PR as is now?

I'm hoping @liuhang-bit can provide input to replace the section with the obsolete APIs with the new ones. The rest of the document is good to go.

File: doc/developer-guides/hld/hld-devicemodel.rst (hunk 1 above)
Comment (amyreye) [reply doc/developer-guides/hld/hld-devicemodel.rst:478]: @liuhang-bit Please add Doxygen comments to vmmapi.h, so we can pull the comments into this hld.

Comment (amyreye) [reply doc/developer-guides/hld/hld-devicemodel.rst:478]: Per offline discussion with @dbkinder @NanlinXie @liuhang-bit, we will resolve this issue soon after the 2.7 release. Issue: https://github.com/projectacrn/acrn-hypervisor/issues/6918. For now, please merge as is for 2.7.

**Review by dbkinder — APPROVED** [review]

**Pull request merged**